docs/provider.adoc

```diff
@@ -0,0 +1,332 @@
+= Arquillian Algeron Pact Provider
```

Comment:
As discussed in the chat - maybe we don't use "Arquillian Algeron" prefix, as it's documentation is for "Arquillian Algeron" anyway :)